### DISCUSSION TITLE
Is it possible to access the Input vector or Last layer of deepmd Neural network?

### DISCUSSION THREAD

**Discussion by rajnichahal** [deepmodeling/deepmd-kit#1852] [General]
Hi, I am wondering if it is possible to access the Input vector or Last layer of the network?

Specifically, I am interested in the information that goes into the first layer of Fitting net (output of Feature matrix) and the last layer of Fitting net (just before per atom total energy, i.e., local energy is calculated).

Thanks!

**Comment by njzjz** [comment]
I think you mean [eval](https://docs.deepmodeling.com/projects/deepmd/en/v2.1.3/api_py/deepmd.infer.html#deepmd.infer.DeepPot.eval) and [eval_descriptor](https://docs.deepmodeling.com/projects/deepmd/en/v2.1.3/api_py/deepmd.infer.html#deepmd.infer.DeepPot.eval_descriptor).

**Reply by rajnichahal** [reply]
Hi, thanks for your response. It looks very similar to what I have been looking for. But I have following questions:
1. Is Python API only available for the newer DeePMD version v.2?
2. Also, looking at the output of 'eval', it seems it is going to give me 1 energy value per configuration/frame (addition of energy from each atom in that frame). Is it possible to get the 'energy of each atom in the frame' for all atoms before it gets added up?
3. I was looking in the deepmd user guide, is this something that I should refer to in order to get the feature matrix for each atom?[https://docs.deepmodeling.com/projects/deepmd/en/v2.1.3/api_py/deepmd.descriptor.html](url)

I am actually interested in visualizing the high-dimensional data from the DeePMD neural network. 
Thanks!

**Reply by njzjz** [reply]
1. `eval_descriptor` is a new method.
2. You may set `atomic` to `True`.
3. I think you mean `eval_descriptor`?

**Reply by rajnichahal** [reply]
Hi, thank you. Yes, I meant eval_descriptor. 
My object oriented programming is a little weak. Looks like the eval is only taking three arguments namely: coord, cell, atype.

Can you please help with setting the atomic to True?

**Reply by njzjz** [reply]
I don't know what you mean, as the API documentation clearly shows the `atomic` argument.

**Reply by rajnichahal** [reply]
Oops. Just found it. Thanks!
